## Sampling Distribution Part B

### Introduction

1. Understanding the concept of Confidence Interval.
2. Estimating the population mean using bootstrapping method.


In [69]:
# Importig libariries and packages
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats
import math
import seaborn as sns
import os
import pandas as pd

In [70]:
#Setting the working directory

os.chdir(r"C:\Users\kusum\Desktop\Python")

In [71]:
# The dataset attrition is read and stored as df

df= pd.read_csv("attrition.csv")  

## Question :- 1
#### a. Consider entire variable as population and calculate mean.

In [72]:
# Calculating mean of population i.e Monthly Income variable

Mean_ppl=np.mean(df.MonthlyIncome)
print("Mean of population: ",Mean_ppl)

Mean of population:  6502.931292517007


#### b. Select the sample S  of size 200 from the population and calculate a 95 % confidence interval estimate of population mean.

In [73]:
# Creating a sample S of size 200 

S=df.MonthlyIncome.sample(200,random_state=1)

# Mean of sample
Mean_S=np.mean(S)
print("Mean of Sample: ",Mean_S)

# Standard error of sample
Std_error=np.std(S)/np.sqrt(200)
print("Standard Error of Sample: ",Std_error)

# Computing the 95% confidence interval
conf_int_a=Mean_S-(1.96*(np.std(S)/np.sqrt(200)))
conf_int_b=Mean_S+(1.96*(np.std(S)/np.sqrt(200)))
print('95% confidence interval =' , conf_int_a,',',conf_int_b)


Mean of Sample:  6496.57
Standard Error of Sample:  338.1411233871148
95% confidence interval = 5833.813398161255 , 7159.326601838745


The Mean of the sample is 6496.57 and it is captured in the 95% confidence interval 5833.813 to 7159.326

#### c. Repeat step 2 ,1000 times and check how many of those Confidence interval's captures the true population mean.

In [75]:
# For counting the no of times mean is captured correctly
count=0                              
for i in range(0,1000):               
    
    # Creating a sample D of size 200 
    D=df.MonthlyIncome.sample(200)

    # Mean of sample
    Mean_D=np.mean(D)

    # Computing the 95% confidence interval
    conf_int_a=Mean_D-(1.96*(np.std(D)/np.sqrt(200)))
    conf_int_b=Mean_D+(1.96*(np.std(D)/np.sqrt(200)))
    
    # Checking if 95% of times the values capture the mean
    if Mean_S >=conf_int_a and Mean_S<=conf_int_b:
        count=count+1
print('The number of times mean is captured correctly ',count)
print('The number of times mean is not-captured correctly ',1000-count)





The number of times mean is captured correctly  953
The number of times mean is not-captured correctly  47


It is observed that in 95% confidence interval, the population estimate is captured 95 times out of 100 and not captured the remaining 5 times.

## Question 2: Wide vs Narrow confidence interval.

#### a. Collect a sample of size 200 from the population and calculate:-
#####                                   1. 90% C.I
#####                                   2. 99% C.I
#### estimates of the population mean.

#### Which one is a wider estimate, state reason.

In [78]:
# Creating a sample S of size 200 

S=df.MonthlyIncome.sample(200,random_state=1)

# Mean of sample
Mean_S=np.mean(S)

# Computing the 90% confidence interval
CI_90_a=Mean_S-(1.645*(np.std(S)/np.sqrt(200)))
CI_90_b=Mean_S+(1.645*(np.std(S)/np.sqrt(200)))
print('90% confidence interval =' , CI_90_a,',',CI_90_b)


# Computing the 99% confidence interval
CI_99_a=Mean_S-(2.575*(np.std(S)/np.sqrt(200)))
CI_99_b=Mean_S+(2.575*(np.std(S)/np.sqrt(200)))
print('99% confidence interval =' , CI_99_a,',',CI_99_b)

90% confidence interval = 5940.327852028196 , 7052.812147971804
99% confidence interval = 5625.856607278179 , 7367.283392721821


The range of 95% C.I is 7052.812147971804 - 5940.327852028196 = 1112.48429594
The range of 99% C.I is 7367.283392721821 - 5625.856607278179 = 1741.42678544

It is clear that 99% C.I is a wider estimate of population mean.

Reason: Bigger the confidence level, wider the interval is, coz with a 90 percent confidence interval, we have a 10 percent chance of missing the estimate of population. And in  99 percent confidence interval we just have 1 percent chance of missing the estimate of population, that mean's it covers the estimates most number of times and is thus wider.

#### b. Collect 2 samples ffrom the population. S1 of size 200 and S2 of size 400 using both the samples calculate 90% C.I estiamtes of mean. Which one is narrower  ? State a reason.

In [80]:
# Creating a sample S1 of size 200 

S1=df.MonthlyIncome.sample(200,random_state=1)

# Creating a sample S2 of size 400 

S2=df.MonthlyIncome.sample(400,random_state=1)

# Mean of sample S1
Mean_S1=np.mean(S1)

# Mean of sample S2
Mean_S2=np.mean(S2)

# Computing the 90% confidence interval for S1
CI_S1_a=Mean_S1-(1.645*(np.std(S1)/np.sqrt(200)))
CI_S1_b=Mean_S1+(1.645*(np.std(S1)/np.sqrt(200)))
print('90% confidence interval for S1=' , CI_S1_a,',',CI_S1_b)

# Computing the 90% confidence interval for S2
CI_S2_a=Mean_S2-(1.645*(np.std(S2)/np.sqrt(200)))
CI_S2_b=Mean_S2+(1.645*(np.std(S2)/np.sqrt(200)))
print('90% confidence interval for S2=' , CI_S2_a,',',CI_S2_b)

90% confidence interval for S1= 5940.327852028196 , 7052.812147971804
90% confidence interval for S2= 5953.562653352553 , 7058.487346647446


Range of C.I for S1 is 1112
Range of C.I for S2 is 1105

The Sample with a bigger size (S2 of size 400) has a narrow range of confidence interval.

Reason: As the sample size increases the value of standard error decreses, which narrows down the range of confidence interval.